src/document_graph/document_graph.cpp

```diff
@@ -105,6 +105,29 @@ namespace hypha
         }
     }
 
+    bool DocumentGraph::hasEdges(const eosio::checksum256 &node)
+    {
+        Edge::edge_table e_t(m_contract, m_contract.value);
+        
+        auto from_node_index = e_t.get_index<eosio::name("fromnode")>();
+        auto from_itr = from_node_index.find(node);
+
+        while (from_itr != from_node_index.end() && from_itr->to_node == node)
+        {
+            return true;
+        }
+
+        auto to_node_index = e_t.get_index<eosio::name("tonode")>();
+        auto to_itr = to_node_index.find(node);
+
+        while (to_itr != to_node_index.end() && to_itr->to_node == node)
+        {
+            return true;
+        }
+
+        return false;
```

Comment:
This can be simplified to 
```
auto from_node_index = e_t.get_index<eosio::name("fromnode")>();
auto to_node_index = e_t.get_index<eosio::name("tonode")>();
auto from_itr = from_node_index.find(node);
auto to_itr = to_node_index.find(node);
return from_itr != from_node_index.end() || to_itr != to_node_index.end();
```

I don't see any reason to use loops, and also conditions doesn't really make sense :p 
Lines 113-115 says: Check if there is an edge which 'from' node is X and 'to' node is also X
Line 121-123 says: Check if there is an edge which 'to' node is X and again 'to' node is X

Reply:
lol. Yes current implementation is terrible. I'll update it.

Reply:
Updated, and fixed a bug that you pointed out in my copy/paste :)